### PR TITLE
[FIX] Crafting rusty shovel causes popup to remove workbench

### PR DESCRIPTION
**Pull request by AzaroQL**
# Description

Main issue:
After crafting rusty shovel click in any part of the window triggers workbench removal modal.
chrome_nEnmf5loq4

Also fixed:
- Cleaned up styles(tailwind classnames).
- Standardized building components(used inside wrapper in all cases).
- Simplified click event handling

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Manual test + yarn test

# Checklist:

- [x] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [ ] Screenshot if it includes any UI changes
- [x] I have read the contributing guidelines and agree to the T&Cs
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by spencerdezartsmith — CHANGES_REQUESTED** [review]
Nice fix and refactor. Just a few comments to fix up. Tag me when you are ready for me to check again 🚀

File: src/features/island/buildings/components/building/Building.tsx [hunk 1]
```diff
@@ -148,42 +148,24 @@ export const Building: React.FC<Prop> = ({
...
+          isBuilt={true}
+          onRemove={hasRustyShovelSelected ? handleRemove : undefined}
+        />
       )}
-      <Modal
-        show={showRemoveModal}
-        centered
-        onHide={() => setShowRemoveModal(false)}
-      >
-        {showRemoveModal && (
```
[the remark this answers is not in the document]
Comment (AzaroQL) [reply src/features/island/buildings/components/building/Building.tsx:179]: fixed

File: src/features/island/buildings/components/building/tent/Tent.tsx [hunk 2]
```diff
@@ -1,18 +1,32 @@
...
+import { BuildingImageWrapper } from "../BuildingImageWrapper";
+
+export const Tent: React.FC<BuildingProps> = ({ isBuilt, onRemove }) => {
+  const handleClick = () => {
+    if (onRemove) {
+      onRemove();
+      return;
+    }
+
+    if (isBuilt) {
```
[the remark this answers is not in the document]
Comment (AzaroQL) [reply src/features/island/buildings/components/building/tent/Tent.tsx:16]: added

File: src/features/island/buildings/components/building/Building.tsx [hunk 3]
```diff
@@ -148,42 +148,24 @@ export const Building: React.FC<Prop> = ({
...
-            <BuildingPlaced
-              buildingId={buildingId}
-              craftingState={building.crafting}
-            />
-          </div>
-        </div>
+        <BuildingPlaced
+          buildingId={building.id}
+          craftingState={building.crafting}
+          isBuilt={true}
```
[the remark this answers is not in the document]
Comment (AzaroQL) [reply src/features/island/buildings/components/building/Building.tsx]: done

**Comment by AzaroQL** [comment]
Addressed review comments and made rebase. @spencerdezartsmith

**Review by spencerdezartsmith — APPROVED** [review]
Excellent job @AzaroQL

**Pull request merged**